mysql-test/suite/innodb/t/general_ts_encrypt.test

```diff
@@ -0,0 +1,266 @@
+			SUBPARTITION p23 TABLESPACE ts_encrypted_new),
+		PARTITION p3 VALUES LESS THAN (60) TABLESPACE ts_encrypted_new (
+			SUBPARTITION p31 TABLESPACE ts_encrypted,
+			SUBPARTITION p32 TABLESPACE innodb_file_per_table,
+			SUBPARTITION p33 TABLESPACE ts_encrypted_new));
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE spt2 ADD PARTITION (PARTITION p4 VALUES LESS THAN (130) (
+			SUBPARTITION p41 TABLESPACE ts_encrypted,
+			SUBPARTITION p42 TABLESPACE ts_unencrypted,
+			SUBPARTITION p43 TABLESPACE ts_encrypted_new));
+
+ALTER TABLE spt2 ADD PARTITION (PARTITION p5 VALUES LESS THAN (140) (
+			SUBPARTITION p51 TABLESPACE ts_encrypted,
+			SUBPARTITION p52 TABLESPACE ts_encrypted_new,
+			SUBPARTITION p53 TABLESPACE ts_encrypted_new));
+
+ALTER TABLE spt2 ADD PARTITION (PARTITION p6 VALUES LESS THAN (150) TABLESPACE ts_unencrypted (
+			SUBPARTITION p61 TABLESPACE ts_encrypted,
+			SUBPARTITION p62 TABLESPACE ts_encrypted_new,
+			SUBPARTITION p63 TABLESPACE ts_encrypted_new));
+
+--error ER_MIX_HANDLER_ERROR
+ALTER TABLE spt2 ADD PARTITION (PARTITION p7 VALUES LESS THAN (160) (
+			SUBPARTITION p71 ENGINE=MyISAM,
+			SUBPARTITION p72 ENGINE=MyISAM,
+			SUBPARTITION p73 ENGINE=MyISAM));
+
+
+# see bug https://bugs.mysql.com/bug.php?id=86809
+
+# --error ER_ILLEGAL_HA_CREATE_OPTION
+# CREATE TABLE pt3 (a INT NOT NULL, PRIMARY KEY(a))
+# 	ENGINE=InnoDB TABLESPACE ts_encrypted ENCRYPTION='y'
+# 	PARTITION BY RANGE (a) PARTITIONS 3 (
+# 		PARTITION p1 VALUES LESS THAN (2),
+# 		PARTITION p2 VALUES LESS THAN (4) TABLESPACE innodb_file_per_table,
+# 		PARTITION p3 VALUES LESS THAN (6) TABLESPACE innodb_system);
+
+# --error ER_ILLEGAL_HA_CREATE_OPTION
+# CREATE TABLE pt4 (a INT NOT NULL, PRIMARY KEY(a))
+# 	ENGINE=InnoDB TABLESPACE ts_encrypted ENCRYPTION='y'
+# 	PARTITION BY RANGE (a) PARTITIONS 3 (
+# 		PARTITION p1 VALUES LESS THAN (2),
+# 		PARTITION p2 VALUES LESS THAN (4) TABLESPACE innodb_file_per_table,
+# 		PARTITION p3 VALUES LESS THAN (6) TABLESPACE ts_unencrypted);
+
+# --error ER_ILLEGAL_HA_CREATE_OPTION
+# CREATE TABLE pt5 (a INT NOT NULL, PRIMARY KEY(a))
+# 	ENGINE=InnoDB TABLESPACE ts_unencrypted ENCRYPTION='y'
+# 	PARTITION BY RANGE (a) PARTITIONS 3 (
+# 		PARTITION p1 VALUES LESS THAN (2),
+# 		PARTITION p2 VALUES LESS THAN (4) TABLESPACE innodb_file_per_table,
+# 		PARTITION p3 VALUES LESS THAN (6) TABLESPACE ts_encrypted);
+
+# SELECT FILE_ID, FILE_NAME FROM INFORMATION_SCHEMA.FILES
+# WHERE FILE_TYPE = 'TABLESPACE' ORDER BY FILE_ID;
+
+# SELECT NAME, FLAG, SPACE_TYPE FROM INFORMATION_SCHEMA.INNODB_SYS_TABLESPACES
+# ORDER BY NAME;
+
+
+# test that innodb_strict_mode=OFF produce error for incorrect encryption
+# options
+SET GLOBAL innodb_file_per_table=OFF;
+SET SESSION innodb_strict_mode=OFF;
+
+CREATE TABLE q1 (c1 int);
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE q1 ENCRYPTION='Y',ALGORITHM=COPY;
+
+eval SET GLOBAL innodb_file_per_table=$innodb_file_per_table;
+eval SET SESSION innodb_strict_mode=$innodb_strict_mode;
```

Comment:
Aren't you setting a session variable to saved global variable here? In any case there is no need to cleanup session variables